### ISSUE TITLE
force:auth:web:login for already-auth'ed user but different alias doesn't update alias (recent regression?)

### ISSUE THREAD

**Issue by SCWells72** [forcedotcom/cli#1031]
I had a user report a behavior in IC2 where he added an OAuth connection which uses `force:auth:web:login` under the hood and that username/alias combination wasn't available afterward. After digging into the logs to see the embedded execution of `force:auth:web:login` followed by `force:org:list`, I was able to see that the provided username was in the list from the latter, but it was associated with a different alias than the one provided to `force:auth:web:login`. I then reproduced this locally on `sfdx-cli/7.103.0 win32-x64 node-v14.17.0` by doing the following:

```
sfdx force:auth:web:login -a alias1 # Log in successfully via browser with username 'foo@bar.baz'
sfdx force:org:list                 # shows 'foo@bar.baz' as having alias 'alias1'
sfdx force:auth:web:login -a alias2 # Log in successfully via browser with username 'foo@bar.baz'
sfdx force:org:list                 # still shows 'foo@bar.baz' as having alias 'alias1' when it should be 'alias2'
```

Please let me know if you need more details.

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-9351642](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000NQzoYAG/view)

**Comment by mshanemc** [comment]
fixed in 7.107.0 https://github.com/forcedotcom/cli/blob/main/releasenotes/README.md#71070-june-24-2021
